blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1010,97 +1056,67 @@ async fn sync_messages_check_state<DB: BlockStore + Send + Sync + 'static, C: Co
-                let compacted_messages = network
-                    .chain_exchange_messages(None, tipset.key(), batch_size as u64)
-                    .await
-                    .map_err(TipsetRangeSyncerError::NetworkMessageQueryFailed)?;
-                // Chain current tipset with iterator
-                let mut inner_iter = std::iter::once(tipset).chain(&mut tipset_iter);
-
-                // Since the bundle only has messages, we have to put the headers in them
-                for messages in compacted_messages {
-                    // Construct full tipset from fetched messages
-                    let tipset = inner_iter.next().ok_or_else(|| {
-                        TipsetRangeSyncerError::NetworkMessageQueryFailed(String::from(
-                            "Messages returned exceeded tipsets in chain",
-                        ))
-                    })?;
-
-                    let bundle = TipsetBundle {
-                        blocks: tipset.blocks().to_vec(),
-                        messages: Some(messages),
-                    };
 
-                    let full_tipset = FullTipset::try_from(&bundle)
-                        .map_err(TipsetRangeSyncerError::GeneratingTipsetFromTipsetBundle)?;
-
-                    // Validate the tipset and the messages
-                    let timer = metrics::TIPSET_PROCESSING_TIME.start_timer();
-                    let current_epoch = full_tipset.epoch();
-                    validate_tipset::<_, C>(
-                        consensus.clone(),
-                        state_manager.clone(),
-                        chainstore.clone(),
-                        bad_block_cache.clone(),
-                        full_tipset,
-                        genesis.clone(),
-                        invalid_block_strategy,
-                    )
-                    .await?;
-                    tracker.write().await.set_epoch(current_epoch);
-                    timer.observe_duration();
-                    metrics::LAST_VALIDATED_TIPSET_EPOCH.set(current_epoch as u64);
-
-                    // Persist the messages in the store
-                    if let Some(m) = bundle.messages {
-                        forest_chain::persist_objects(chainstore.blockstore(), &m.bls_msgs)?;
-                        forest_chain::persist_objects(chainstore.blockstore(), &m.secp_msgs)?;
-                    } else {
-                        warn!("ChainExchange request for messages returned null messages");
-                    }
-                }
+            // Request tipset messages via chain_exchange
+            batch.push(tipset);
+            if batch.len() == REQUEST_WINDOW {
+                fetch_batch(&batch, &network, &task_chainstore, &s).await?;
+                batch.clear();
             }
         }
+        // Fetch last batch
+        fetch_batch(&batch, &network, &task_chainstore, &s).await?;
+        batch.clear();
+
+        Ok::<(), TipsetRangeSyncerError<C>>(())
```

Comment:
Can `rustc` handle this on its own?
```suggestion
        Ok(())
```

Reply:
Done.